notificationworkerlambda/cdk/lib/senderworker.ts

```diff
@@ -115,15 +117,37 @@ class SenderWorker extends cdkcore.Construct  {
       reservedConcurrentExecutions: props.reservedConcurrency
     })
 
-    const senderSqsEventSourceMapping = new lambda.EventSourceMapping(this, "SenderSqsEventSourceMapping", {
-      batchSize: props.isBatchingSqsMessages ? 20 : 1,
-      maxBatchingWindow: props.isBatchingSqsMessages ? cdk.Duration.seconds(1) : cdk.Duration.seconds(0),
+    const eventRule = new Rule(this, "SqsMessageSentRule", {
       enabled: true,
-      eventSourceArn: this.senderSqs.queueArn,
-      target: senderLambdaCtr
-    })
-    senderSqsEventSourceMapping.node.addDependency(this.senderSqs)
-    senderSqsEventSourceMapping.node.addDependency(senderLambdaCtr)
+      description: `SQS message sent to SQS queue for ${props.platform} on ${scope.stage}`,
+      ruleName: `${scope.stack}-sender-${props.platform}-${scope.stage}-trigger`,
+      eventPattern: {
+        source: ["com.gu.notifications.worker.Harvester"],
+        detailType: ["Tokens sent"],
+        detail: {
+          platform: [`${props.platform}`],
+        }
+      },
+    });
+    eventRule.addTarget(new LambdaFunction(senderLambdaCtr, {
+      maxEventAge: cdk.Duration.minutes(15), // Optional: set the maxEventAge retry policy
+      retryAttempts: 5, // Optional: set the max number of retry attempts
+    }));
+
+    const scheduledEventRule = new Rule(this, "ScheduledRerunRule", {
+      schedule: Schedule.cron({ minute: "0/10" }),
+    });
+    scheduledEventRule.addTarget(new LambdaFunction(senderLambdaCtr, {
+      event: RuleTargetInput.fromObject({
+        source: "com.gu.notifications.worker",
+        "detail-type": "Scheduled sender",
+        resources: [`${this.senderSqs.queueUrl}`],
+        detail: {
+          platform: `${props.platform}`,
+          batchsize: 10,
```

Comment:
Is this the same for both ios and android? When we were testing previously the ios lambda didn't process a "big" batch size very well. Given the lambda code (and the underlying concurrency model?) is unchanged as a result of this PR I suppose we could see a similar degradation in  behaviour when processing lots of SQS messages per invocation. It's worth testing this though because the result may be different :)

Reply:
Oh, sorry, maybe this is a different batch size? (I was thinking of the sqs message batch size that the lambda uses to get messages from the queue)

Reply:
It is a different batch size.  It is used in the scheduled execution of the sender lambda to process SQS messages which are left on the queue for some reasons. 

For invocation that are triggered by the EventBridge event (sent by Harvester), the lambda gets the batch size from a new SSM parameter `batchsize`.  It is set to 10 for Android (the max batch size allowed in `receiveMessage` API) and set to 1 for the rest (to be in line with our current settings).